### PR TITLE
Address most of Marie-Jose's comments

### PR DESCRIPTION
**Pull request by brunorijsman**

### CODE REVIEW

**Comment by Wojtek242** [comment]
Thanks a lot @brunorijsman! I haven't looked at the content changes yet, but one of your commits (I believe the second one) completely reformatted the indentation of the entire xml file which I'd rather not do as my editor will probably insist on undoing all of that. Would you be able to revert the indentation changes for this PR?

**Comment by brunorijsman** [comment]
Urghh… I will undo that.  I thought I had turned off Visual Studio Code’s “auto format on save” but evidently not.

**Comment by brunorijsman** [comment]
Fixed it.

File: draft-irtf-qirg-principles-09-rc.xml
Comment (Wojtek242) [inline draft-irtf-qirg-principles-09-rc.xml]: I would not capitalize "internet" here since it's "**a** quantum internet"

File: draft-irtf-qirg-principles-09-rc.xml
Comment (Wojtek242) [inline draft-irtf-qirg-principles-09-rc.xml:225]: I'm wondering whether this should also be included: https://www.science.org/doi/10.1126/sciadv.aba0959. It's also entanglement based, but without quantum memories or swapping.

File: draft-irtf-qirg-principles-09-rc.xml
Comment (Wojtek242) [inline draft-irtf-qirg-principles-09-rc.xml]: I'd also add secure delegated quantum computing (i.e. blind quantum computing).

File: draft-irtf-qirg-principles-09-rc.xml
Comment (Wojtek242) [inline draft-irtf-qirg-principles-09-rc.xml]: I think the better solution is to just avoid mentioning entanglement or quantum states until they are defined.

I propose to rephrase this sentence as follows: "The first fully quantum networks capable of transmitting and manipulating distributed quantum information ("prepare and measure" stage in [WEHNER18]) via an intermediate quantum node have recently been realized"

File: draft-irtf-qirg-principles-09-rc.xml
Comment (Wojtek242) [inline draft-irtf-qirg-principles-09-rc.xml]: The Castelvecchi and Wehner references need to be moved or removed. I'd say remove both.

File: draft-irtf-qirg-principles-09-rc.xml
Comment (Wojtek242) [inline draft-irtf-qirg-principles-09-rc.xml]: I don't think this rephrasing reflects the original meaning. I think I'd rephrase to "by the application itself using low-level instructions sets, including quantum assembly (QASM) languages, rather than being managed by a network stack".

File: draft-irtf-qirg-principles-09-rc.xml
Comment (Wojtek242) [inline draft-irtf-qirg-principles-09-rc.xml:318]: Quantum state is more than that, but it's a good point, there probably needs to be some text that explains what a quantum state is and how it is different and relates to a qubit.

File: draft-irtf-qirg-principles-09-rc.xml
Comment (Wojtek242) [inline draft-irtf-qirg-principles-09-rc.xml]: Perhaps, "It expresses the probability that the state will behave exactly the same as our desired state"? Yielding the same result is not necessarily correct I think as that will depend on the measurement.

File: draft-irtf-qirg-principles-09-rc.xml
Comment (Wojtek242) [inline draft-irtf-qirg-principles-09-rc.xml]: Should probably also add a reference to the generations paper "muhlaridan" I think...

File: draft-irtf-qirg-principles-09-rc.xml
Comment (Wojtek242) [inline draft-irtf-qirg-principles-09-rc.xml:594]: Can make a reference to the generations paper again and also refer to the section where the generations are explained.

File: draft-irtf-qirg-principles-09-rc.xml
Comment (Wojtek242) [inline draft-irtf-qirg-principles-09-rc.xml:815]: Is this normal/preferred? It won't be a draft after it's published, no?

File: draft-irtf-qirg-principles-09-rc.xml
Comment (Wojtek242) [inline draft-irtf-qirg-principles-09-rc.xml]: I'd just say they are necessarily "stateful" without calling it "stateful distributed computation. Whilst I know some like this phrasing, I know people who will disagree with it (and I'm neutral on this). The document is meant to be a consensus so I'd avoid it.

File: draft-irtf-qirg-principles-09-rc.xml
Comment (Wojtek242) [inline draft-irtf-qirg-principles-09-rc.xml:1314]: We can also add "trapped ions [REF: https://www.nature.com/articles/nature06118]" reference for better coverage.

File: draft-irtf-qirg-principles-09-rc.xml
Comment (Wojtek242) [inline draft-irtf-qirg-principles-09-rc.xml:1397]: 10 Hz seems a bit random without a reference.

File: draft-irtf-qirg-principles-09-rc.xml
Comment (Wojtek242) [inline draft-irtf-qirg-principles-09-rc.xml]: That's not strictly correct. Teleportation can indeed do it, but if the state you need can be constructed from Bell states then you don't need teleportation. I would rephrase this a bit.

I would state that "...quantum applications consume entanglement like a resource..." and then state that this fundamental resource can be either used directly in an application, e.g. QKD which simply measures the entangled qubits to obtain a secret key, used to build more complex abstraction and operations, e.g. distributed CNOT gates (https://www.quantum-network.com/applications/distributed-cnot/),  or to directly teleport quantum information.

File: draft-irtf-qirg-principles-09-rc.xml
Comment (Wojtek242) [reply draft-irtf-qirg-principles-09-rc.xml:225]: No need to act on this, leaving this for the QIRG mailing list.

File: draft-irtf-qirg-principles-09-rc.xml
Comment (Wojtek242) [reply draft-irtf-qirg-principles-09-rc.xml]: Should also add references to these as well. If you feel like it, you can have a look at the intro in https://dl.acm.org/doi/pdf/10.1145/3386367.3431293 and add them. Feel free, to leave this alone as well and I'll add them later.

**Review by Wojtek242 — CHANGES_REQUESTED** [review]

**Comment by Wojtek242** [comment]
See also e-mail on QIRG mailing list: https://mailarchive.ietf.org/arch/msg/qirg/TNPvzu90ImcvvCXhT__clK6xbSc/

File: draft-irtf-qirg-principles-09-rc.xml
Comment (brunorijsman) [reply draft-irtf-qirg-principles-09-rc.xml]: Done: changed Internet to internet (lower case i)

File: draft-irtf-qirg-principles-09-rc.xml
Comment (brunorijsman) [reply draft-irtf-qirg-principles-09-rc.xml:225]: Interesting paper; thanks for pointing to it. I can see how the approach described in this paper allows for a rich topology without using memories or swapping, but I do not see how it overcomes the distance limitations of existing QKD (~ 120 km for BB84, ~ 600 km for TF).  The "long distances" referred to in this paper (start of page 4) are only up to 12.6 km.  Anyway... since you said "no need to act on this, leaving it to the mailing list" I am marking the comment as resolved.

File: draft-irtf-qirg-principles-09-rc.xml
Comment (brunorijsman) [reply draft-irtf-qirg-principles-09-rc.xml]: Added "blind quantum computing" and "clock synchronization". I will leave it to you to add the references.

File: draft-irtf-qirg-principles-09-rc.xml
Comment (brunorijsman) [reply draft-irtf-qirg-principles-09-rc.xml]: Did you notice that I added a reference to [Pompili21] (https://arxiv.org/pdf/2102.04471.pdf).  It seems to me that this goes beyond the "prepare and measure" stage.

File: draft-irtf-qirg-principles-09-rc.xml
Comment (brunorijsman) [reply draft-irtf-qirg-principles-09-rc.xml]: I have removed the references to Castelvecchi and Wehner.  Note that this contradicts your previous comment where you suggested changing the sentence to "The first fully quantum networks capable of transmitting and manipulating distributed quantum information ("prepare and measure" stage in [WEHNER18]) via an intermediate quantum node have recently been realized" which *does* include the Wehner reference.

File: draft-irtf-qirg-principles-09-rc.xml
Comment (brunorijsman) [reply draft-irtf-qirg-principles-09-rc.xml:815]: I'm with you, Woj. But Marie-Jose's response was firm on this. Leaving "draft".

File: draft-irtf-qirg-principles-09-rc.xml
Comment (brunorijsman) [reply draft-irtf-qirg-principles-09-rc.xml]: Updated according to your suggestion.

File: draft-irtf-qirg-principles-09-rc.xml
Comment (brunorijsman) [reply draft-irtf-qirg-principles-09-rc.xml:1314]: No objection to adding the reference, but what do you suggest we say single-atom qubits to contrast them with atomic ensembles and NV-center?

File: draft-irtf-qirg-principles-09-rc.xml
Comment (brunorijsman) [reply draft-irtf-qirg-principles-09-rc.xml:1397]: I copied the 10 Hz from the paragraph right above it, which was already there and which does have a reference. I double checked, and the reference actually has the number 10Hz in it (figures 17 through 22 and table 3 all have throughputs peaking below 10 Hz)

File: draft-irtf-qirg-principles-09-rc.xml
Comment (brunorijsman) [reply draft-irtf-qirg-principles-09-rc.xml]: Updated the sentence as you suggested.  Also added the reference to the QNE distributed CNOT page (as well as another reference to the QNE teleportation page).

File: draft-irtf-qirg-principles-09-rc.xml
Comment (Wojtek242) [reply draft-irtf-qirg-principles-09-rc.xml]: Thanks, I'll add them after merging your PR

File: draft-irtf-qirg-principles-09-rc.xml
Comment (Wojtek242) [reply draft-irtf-qirg-principles-09-rc.xml]: Yes, I've seen. But I was just trying to define "fully quantum" network without mentioning entangled states and I think that defining it as "prepare and measure" stage (which defines minimum end-node capabilities) via an intermediate node (so that it's not a p2p link and actually some sort of network) is suitable. And yes, the Delft network is quantum memory stage anyway, but I don't think we have to draw the line that high for the purposes of this draft.

File: draft-irtf-qirg-principles-09-rc.xml
Comment (Wojtek242) [reply draft-irtf-qirg-principles-09-rc.xml]: Haha yes. I meant kick out the ones at the end that were placed there originally as a reference relative to "imminently realized" to being relative to the mention of quantum network stages.

File: draft-irtf-qirg-principles-09-rc.xml
Comment (Wojtek242) [reply draft-irtf-qirg-principles-09-rc.xml:815]: Ack

File: draft-irtf-qirg-principles-09-rc.xml
Comment (Wojtek242) [reply draft-irtf-qirg-principles-09-rc.xml:1314]: Single-atom qubits is too broad as that could also be neutral atoms.

File: draft-irtf-qirg-principles-09-rc.xml
Comment (Wojtek242) [reply draft-irtf-qirg-principles-09-rc.xml:1397]: Ah yes, that makes sense. Silly GitHub hiding the paragraph above.

File: draft-irtf-qirg-principles-09-rc.xml
Comment (brunorijsman) [reply draft-irtf-qirg-principles-09-rc.xml]: I took the easy way out: mention neither entanglement nor prepare-and-measurement at this early stage in the document:

"The first fully quantum networks capable of sending, receiving, and manipulating distributed quantum information have recently been realized [Pompili21]."

Pushed another commit with this change.

File: draft-irtf-qirg-principles-09-rc.xml
Comment (brunorijsman) [reply draft-irtf-qirg-principles-09-rc.xml:1314]: Yeah, sorry typo. I meant to say "ion-trap" instead of"single-atom". Anyway, can you suggest some specific language to add?

File: draft-irtf-qirg-principles-09-rc.xml
Comment (brunorijsman) [reply draft-irtf-qirg-principles-09-rc.xml]: Restored the language that you suggested. All added a reference to the OpenQASM 3.0 paper.

File: draft-irtf-qirg-principles-09-rc.xml
Comment (brunorijsman) [reply draft-irtf-qirg-principles-09-rc.xml]: Don't you think this may be splitting the hair too finely for a classical network engineer and making the explanation less clear rather than more clear the process?  I thought it was rather obvious from the context that we are doing the same measurement on the desired state and the target state... Anyway, if you feel strongly about it, I can change the language to what you suggested.

File: draft-irtf-qirg-principles-09-rc.xml
Comment (brunorijsman) [reply draft-irtf-qirg-principles-09-rc.xml]: Happy to add it. However, Googling "muhlaridan generations" (or variations thereof) did not yield anything.  Do you have a link to the paper?

File: draft-irtf-qirg-principles-09-rc.xml
Comment (brunorijsman) [reply draft-irtf-qirg-principles-09-rc.xml:594]: Ditto - Do you have a link for the paper that you have in mind? I can add it (and the reference to the section).

File: draft-irtf-qirg-principles-09-rc.xml
Comment (Wojtek242) [reply draft-irtf-qirg-principles-09-rc.xml]: It's already in this draft as "Mural16". Sorry, I just spelled it terribly bad.

File: draft-irtf-qirg-principles-09-rc.xml
Comment (Wojtek242) [reply draft-irtf-qirg-principles-09-rc.xml]: I think that at the same time it is important to be correct though. I do think that some variation "It expresses the probability that the state will behave exactly the same as our desired state" or "It expresses the probability that the state will **mimic the behaviour of the** desired state" is a good balance between the original phrase (which was paraphrased from wikipedia) and your suggestion which is not correct in all circumstances.

File: draft-irtf-qirg-principles-09-rc.xml
Comment (Wojtek242) [reply draft-irtf-qirg-principles-09-rc.xml:1314]: On the other hand, nitrogen-vacancy <xref target="Hensen15" /> or trapped ion <NEW REF> based hardware offer a much greater degree...

Comment (Wojtek242) [reply draft-irtf-qirg-principles-09-rc.xml:1314]: I wouldn't bother contrasting, just adding to the list so that it's not so centered around NVs

File: draft-irtf-qirg-principles-09-rc.xml
Comment (Wojtek242) [reply draft-irtf-qirg-principles-09-rc.xml:318]: On my TODO list

File: draft-irtf-qirg-principles-09-rc.xml
Comment (brunorijsman) [reply draft-irtf-qirg-principles-09-rc.xml]: I changed the description of fidelity to the first suggestion that you provided.

File: draft-irtf-qirg-principles-09-rc.xml
Comment (brunorijsman) [reply draft-irtf-qirg-principles-09-rc.xml]: Added reference.

File: draft-irtf-qirg-principles-09-rc.xml
Comment (brunorijsman) [reply draft-irtf-qirg-principles-09-rc.xml:594]: Added ref to Mural16 fig 2 here.

File: draft-irtf-qirg-principles-09-rc.xml
Comment (brunorijsman) [reply draft-irtf-qirg-principles-09-rc.xml:1314]: Added mention of ion trap, with reference to Moehring07, without specific pros and cons.

**Comment by brunorijsman** [comment]
Okay, I think I have resolved all comments. We still need to run xml2rfc to generate the new text.

**Comment by Wojtek242** [comment]
Awesome, thanks!

**Comment by Wojtek242** [comment]
Can you run xml2rfc to regenerate the text file. Then I'll just do a "squash and merge" and the update will be all in one commit.

**Comment by brunorijsman** [comment]
I ran xml2rfc, produced the .txt files, and fixed some issues that xml2rfc found in the xml file

Go ahead, squash and merge

**Pull request merged**
